### DISCUSSION TITLE
Font in lables

### DISCUSSION THREAD

**Discussion by IshanJ25** [TomSchimansky/CustomTkinter#152] [General]
i have variable weights of fonts installed on my system, but i cannot access, for example, the `medium` or `black` variant of the font to show in my `ctkframe`. I can only access the regular and bold variants. How can this be resolved?

`    title = ctk.CTkLabel(login_frame, text='Title', text_font='Poppins 20 normal',)` --> works fine

`    title = ctk.CTkLabel(login_frame, text='Title', text_font='Poppins-black 20 normal',)`  --> shows arial font

**Comment by IshanJ25** [comment]
![image](https://user-images.githubusercontent.com/86649457/172017138-eab5b861-09be-4a67-abf4-1e3f4ea69aab.png)

**Comment by felipetesc** [comment]
I'll shoot in the dark here (that's,  I'll need your guidance  afterwards): 

About the image above, is it showing Poppins font type weights? 
Case positive, where is the black weight ?  I can't see on the image.

In order to test and to check font usage I created the code bellow (I have downloaded Roboto, since its available at no cost to Windows, Linux and MacOS, and I've installed system wide):

```python 
import tkinter
import tkinter.font as Font
import customtkinter as ctk

class App(ctk.CTk):

    def __init__(self):
        super().__init__()
        self.title("Test custom font")
        self.resizable(height = 800, width = 600)
        title =  ctk.CTkLabel(self, text='Title', text_font="roboto-extralight 40 italic")
        title.place(anchor="center", relx="0.5", rely="0.5")


a = App()
a.mainloop()
```

To answer the question: I think you are having a problem with the order of the string and I'm guessing, that you got Arial as a result, because the option Poppins-black isn't valid, and tkinter returns the default font so your application doesn't break.

**Comment by IshanJ25** [comment]
I tried this with `roboto black` since i don't have `extralight` installed on my system

```python
import tkinter
import tkinter.font as Font
import customtkinter as ctk

class App(ctk.CTk):

    def __init__(self):
        super().__init__()
        self.title("Test custom font")
        self.resizable(height=800, width=600)
        title = ctk.CTkLabel(self, text='Title', text_font="roboto-black 40 normal")
        title.place(anchor="center", relx="0.5", rely="0.5")


a = App()
a.mainloop()
```

here is what robot black should look like:
![image](https://user-images.githubusercontent.com/86649457/172056894-4f47cc1b-a588-4c5f-8fb4-672711f0103f.png)

and here is the output window:
![image](https://user-images.githubusercontent.com/86649457/172056922-5f14e8c0-a30f-4439-97c4-1d21c85f5a5e.png)

**Reply by felipetesc** [reply]
Yes, it's a bug.

**Reply by felipetesc** [reply]
But it's also related about how ctk reads font files and creates the themes. Inside the file ctk_label.py there is a link to one stackoverflow issue related to tkinter fonts: https://stackoverflow.com/questions/11993290/truly-custom-font-in-tkinter/30631309#30631309. Another thing, is how themes are constructed and saved inside json files, there are font names which specify what fonts should be used for linux, windows and mac. Also, the folder CustomTkinter/customtkinter/assets/fonts/ contains two Roboto files.

I'll keep looking into this to check how to deal with the addition of custom fonts

**Comment by XxDrexX-cmd** [comment]
> 

put in a tuple

**Comment by IshanJ25** [comment]
Any update?
